### DISCUSSION TITLE
(PowerShell 7.5 or higher) replacement version of chkdsk, now known as SysFix for PowerShell only

### DISCUSSION THREAD

**Discussion by EdgeDev01** [PowerShell/PowerShell#20735] [Enhancements]
the command would do the following all at once: 
- Scan and repair corrupted system files.
- Optimize and Defragment the disk space
- Clean up temporary files
- Restore the system image to a healthy state
- Provide detailed reports and logs of every operation
- Scan and Remove any malware or viruses using MSRT and Windows Defender
- Compare and verify the system files with the same Windows Version on a clean install vs a non-clean install
- Detect and fix any discrepancies or inconsistencies in the system files
Maybe the command would look like this: SystemFix /Online /All /Report /Scan /Compare

**Comment by rhubarb-geek-nz** [comment]
I suggest a tool such as this should be its own separate `git` project and published using the [PowerShell Gallery](https://www.powershellgallery.com/) rather than being part of the core `PowerShell`. By it's definition, it is a windows only tool, not a cross platform tool. There is nothing in the description that requires it to be PowerShell 7.5 or higher. Given its role, it should be designed to work with the built-in classic PowerShell 5.1 so a system can be recovered by using the smallest additional download.

Although it is described as a replacement for `chkdsk`, it would need to use chkdsk or some other disk management tool to perform the hard drive management. The existing chkdsk requires a reboot cycle to perform fixes of the boot volume so this is not a simple run this tool once thing.

I suggest it is built as a tool which orchestrates existing tools rather than re-inventing the wheel.

The tool needs to have a very well defined scope otherwise I fear it will never be finished, for instance what are the range of errors and fixes that are in scope so that by definition anything else is out of scope.
